bindings/python/pyproject.toml

```diff
@@ -23,6 +23,7 @@ build-backend = "maturin"
 name = "pyiceberg_core"
 version = "0.4.0"
 readme = "project-description.md"
+requires-python = "~=3.9"
```

Comment:
i think this means only version 3.9. we support `^3.9` in pyiceberg
https://github.com/apache/iceberg-python/blob/b15934d5a9e6bf97b047f63239cc21ba1c15cdd4/pyproject.toml#L52

Reply:
ah youre right 

```
~=3.1: version 3.1 or later, but not version 4.0 or later.
```
https://peps.python.org/pep-0440/#examples